authentik/enterprise/reports/models.py

```diff
@@ -0,0 +1,125 @@
+import csv
+import io
+from uuid import uuid4
+
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.utils.translation import gettext as _
+from rest_framework.serializers import Serializer
+from rest_framework.settings import api_settings
+from rest_framework.viewsets import ModelViewSet
+
+from authentik.admin.files.fields import FileField
+from authentik.admin.files.manager import get_file_manager
+from authentik.admin.files.usage import FileUsage
+from authentik.core.models import User
+from authentik.enterprise.reports.utils import MockRequest
+from authentik.events.models import Event, EventAction, Notification, NotificationSeverity
+from authentik.lib.models import SerializerModel
+from authentik.lib.utils.db import chunked_queryset
+from authentik.tenants.utils import get_current_tenant
+
+
+class DataExport(SerializerModel):
+    id = models.UUIDField(primary_key=True, default=uuid4)
+    requested_by = models.ForeignKey(User, null=True, on_delete=models.SET_NULL)
+    requested_on = models.DateTimeField(auto_now_add=True)
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    query_params = models.JSONField()
+    file = FileField(blank=True)
+    completed = models.BooleanField(default=False)
+
+    class Meta:
+        verbose_name = _("Data Export")
+        verbose_name_plural = _("Data Exports")
+
+    @property
+    def serializer(self) -> type[Serializer]:
+        """Get serializer for this model"""
+        from authentik.enterprise.reports.api.reports import DataExportSerializer
+
+        return DataExportSerializer
+
+    def generate(self) -> None:
+        if self.completed:
+            raise AssertionError("Data export must only be generated once")
```

Comment:
should preferably be an error that inherits from `authentik.lib.sentry.SentryIgnoredException`